### PR TITLE
Remove password field requirement for /users/{id} endpoints

### PR DESCRIPTION
**Pull request by rbassot**
- serializer is updated to accept AppUser data with both a password and a lack of a password
- views logic updated for `CREATE /users` to assert that a password field does exist (required to create a Prof account)
- update the UPDATE /users/{id} test to send password-less data

### CODE REVIEW

**Review by skosman — APPROVED** [review]
Tested the update prof endpoint with no password and it works! Thanks for the speedy fix! Also tested new prof and got the correct error

File: users/views.py [hunk 1]
```diff
@@ -35,8 +35,12 @@ def get(self, request):
     def post(self, request: HttpRequest, format=None) -> HttpResponse:
         if request.method != "POST":
             return HttpResponse(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
-
         request_data = JSONParser().parse(request)
+
+        #upon Prof creation, manually assert that a password was provided in the request body
+        if 'password' not in request_data['user']:
```
Comment (Russell-Waterhouse) [inline users/views.py:41]: If we're getting rid of the password being needed what's the point of this check?

Comment (skosman) [reply users/views.py:41]: You need it to create a professor

Comment (rbassot) [reply users/views.py:41]: This affects only the Admin-side professor _creation_ endpoint. Logically, a Prof account must be created with a password field set, so this is an easy way to allow both cases to happen without messing around with the serializer too much. Makes sense?

Comment (Russell-Waterhouse) [reply users/views.py:41]: Oh silly me

**Review by Russell-Waterhouse — APPROVED** [review]

**Pull request merged**